pyipmi/hpm.py

```diff
@@ -203,7 +203,7 @@ def wait_for_long_duration_command(self, expected_cmd, timeout, interval):
             try:
                 status = self.get_upgrade_status()
                 if status.command_in_progress is not expected_cmd \
-                        and status.command_in_progress is not 0x34:
+                        and status.command_in_progress != 0x34:
```

Comment:
I would fix also `is not` in the above line.

Try to compile Python without the small integers cache and with disabled constants deduplication. I am sure many tests will fail.